lib/provider/drift/chat_item.dart

```diff
@@ -301,6 +302,73 @@ class ChatItemDriftProvider extends DriftProviderBaseWithScope {
 
     return result ?? [];
   }
+
+  /// Returns the [DtoChatItem]s being in a historical view order of the
+  /// provided [chatId].
+  Future<Stream<List<MapChangeNotification<ChatItemId, DtoChatItem>>>> watch(
+    ChatId chatId, {
+    int? before,
+    int? after,
+    PreciseDateTime? around,
+  }) async {
```

Comment:
The current approach WON'T work flawlessly, because when a new item is added, this method must be re-triggered with the new `before` and `after` values, which will completely re-query the SELECT clause. This doesn't sound right. Is this necessary? Yes, because we need to monitor the messages being added inside the page we're watching right now, and if this page expands, then the `watch()` must expand as well, so it doesn't lose any new `ChatItem`s added after or before it.

However, since new items will always be added to the bottom, then perhaps we may just listen to the bottom changes? But the calls might change, the message editing is a thing as well. Thus perhaps we should `watch()` for every page we query, yet the latest one (the bottom one) is always there?